### PR TITLE
Feat: Editor plugins can be rendered without an active document

### PR DESCRIPTION
**Pull request by trusz**
Resolves:
- https://github.com/openscd/open-scd/issues/1512

Pre-requisite:
- https://github.com/openscd/open-scd/pull/1578

Replaces:
- https://github.com/openscd/open-scd/pull/1575

### CODE REVIEW

File: packages/openscd/src/plugins.ts [hunk 1]
```diff
@@ -9,97 +9,111 @@ export const officialPlugins = [
     icon: 'developer_board',
     default: true,
     kind: 'editor',
+    requireDoc: true,
```
Comment (clepski) [inline packages/openscd/src/plugins.ts:12]: Requiring all old plugins to add this property would be a pretty large breaking change, can we default this is property to `true` for editor plugins and only use it if it is explicitly set to `false` instead?

File: packages/openscd/src/addons/Layout.ts [hunk 2]
```diff
@@ -469,17 +469,26 @@ export class OscdLayout extends LitElement {
...
       </mwc-tab-bar>
-      ${renderEditorContent(this.editors, this.activeTab)}
+      ${renderEditorContent(this.editors, this.activeTab, this.doc)}
     `;
 
-    function renderEditorContent(editors: Plugin[], activeTab: number){
-      const content = editors[activeTab]?.content;
+    function renderEditorContent(editors: Plugin[], activeTab: number, doc: XMLDocument | null){
+      const editor = editors[activeTab];
+      const requireDoc = editor?.requireDoc
```
Comment (clepski) [inline packages/openscd/src/addons/Layout.ts:488]: We could assume the default for requireDoc is `true` if missing, e.g. do 
```
const requireDoc = editor?.requireDoc !== false
```

File: packages/openscd/src/addons/Layout.ts [hunk 3]
```diff
@@ -633,6 +642,14 @@ export class OscdLayout extends LitElement {
...
             value="${plugin.src}"
             ?selected=${plugin.installed}
+            @request-selected=${(e: CustomEvent<{source: string}>) => {
+              if(e.detail.source !== 'interaction'){
+                e.preventDefault();
+                e.stopPropagation();
+                e.stopImmediatePropagation();
+                return false;
+              }
+            }}
```
Comment (clepski) [inline packages/openscd/src/addons/Layout.ts:652]: This is required to prevent rerendering?

File: packages/openscd/src/plugins.ts (hunk 1 above)
Comment (trusz) [reply packages/openscd/src/plugins.ts:12]: done in `Layout.ts`

File: packages/openscd/src/addons/Layout.ts (hunk 2 above)
Comment (trusz) [reply packages/openscd/src/addons/Layout.ts:488]: We need the default value also for the tabs so I check earlier.

File: packages/openscd/src/addons/Layout.ts (hunk 3 above)
Comment (trusz) [reply packages/openscd/src/addons/Layout.ts:652]: yes

**Review by clepski — APPROVED** [review]

**Pull request merged**